bsoid_app/video_creator.py

```diff
@@ -129,100 +132,107 @@ def setup(self):
-                     'at {} frames per second'.format(self.num_frames, self.avg_frame_rate)):
-            st.info('Extracting frames from the video... ')
+        print('Extracting frames from videos...')
+        for vid_file_path in self.vid_file:
+            full_vid_path = os.path.join(self.vid_dir, vid_file_path)
             try:
-                (ffmpeg.input(os.path.join(self.vid_dir, self.vid_file))
-                 .filter('fps', fps=self.avg_frame_rate)
-                 .output(str.join('', (self.frame_dir, '/frame%01d.png')), video_bitrate=self.bit_rate,
-                         s=str.join('', (str(int(self.width * 0.5)), 'x', str(int(self.height * 0.5)))),
-                         sws_flags='bilinear', start_number=0)
+                probe = ffmpeg.probe(full_vid_path)
+                video_stream = next((stream for stream in probe['streams'] if stream['codec_type'] == 'video'), None)
+                if video_stream is None:
+                    print(f"No video stream found in {vid_file_path}")
+                    continue
+                avg_frame_rate = eval(video_stream['avg_frame_rate'])
+                num_frames = int(video_stream['nb_frames'])
+                width = int(video_stream['width'])
+                height = int(video_stream['height'])
+                bit_rate = int(video_stream['bit_rate'])
+                frames_dir = os.path.join(self.frame_dir, os.path.splitext(vid_file_path)[0] + '_frames')
+                os.makedirs(frames_dir, exist_ok=True)
+                (ffmpeg
+                 .input(full_vid_path)
+                 .filter('fps', fps=avg_frame_rate)
+                 .output(os.path.join(frames_dir, 'frame%05d.png'), video_bitrate=bit_rate,
+                         s=f"{int(width * 0.5)}x{int(height * 0.5)}", sws_flags='bilinear', start_number=0)
                  .run(capture_stdout=True, capture_stderr=True))
-                st.info('Done extracting **{}** frames from video **{}**.'.format(self.num_frames, self.vid_file))
+                print(f'Done extracting {num_frames} frames from video {vid_file_path} into {frames_dir}')
             except ffmpeg.Error as e:
-                st.error('stdout:', e.stdout.decode('utf8'))
-                st.error('stderr:', e.stderr.decode('utf8'))
-            st.info('Done extracting {} frames from {}'.format(self.num_frames, self.vid_file))
-
+                print(f'Error extracting frames from {vid_file_path}:')
+                print('ffmpeg stdout:', e.stdout.decode('utf8'))
+                print('ffmpeg stderr:', e.stderr.decode('utf8'))
+        print('Finished extracting frames from all videos.')
+    
     def create_videos(self):
-        radio = st.radio(label='Have you extracted frames?', options=["Yes", "No"])
-        if radio == 'Yes':
-            if st.checkbox('Clear old videos? Uncheck after check to prevent from auto-clearing', False, key='vr'):
+        self.frame_extraction()
```

Comment:
Lol this code is so messy (not your fault). Did you do any meaningful changes besides removing `st` stuff?

Reply:
main changes were getting it to create videos for all the input folders vs. the user repeatedly going through this part of the app one file at a time if they wanted it done for multiple files/videos